php/ext/google/protobuf/encode_decode.c

```diff
@@ -612,9 +659,10 @@ static void map_slot_value(upb_fieldtype_t type, const void* from,
 static void *startmapentry_handler(void *closure, const void *hd) {
   MessageHeader* msg = closure;
   const map_handlerdata_t* mapdata = hd;
+  CACHED_VALUE* cache = find_zval_property(msg, mapdata->fd);
   TSRMLS_FETCH();
-  zval* map = CACHED_PTR_TO_ZVAL_PTR(
-      DEREF(message_data(msg), mapdata->ofs, CACHED_VALUE*));
+  map_field_insure_created(mapdata->fd, cache PHP_PROTO_TSRMLS_CC);
```

Comment:
What's the difference between TSRMLS_CC and PHP_PROTO_TSRMLS_CC?

Reply:
PHP_PROTO_TSRMLS_CC is to support both php 5 and php 7.
Some php internal API differ from php 5 and php7.
In order to unify code, I created PHP_PROTO_TSRMLS_CC.
When it's php 5, PHP_PROTO_TSRMLS_CC is TSRMLS_CC
Otherwise, it's empty